src/google/protobuf/compiler/command_line_interface.cc

```diff
@@ -188,8 +188,25 @@ bool TryCreateParentDirectory(const string& prefix, const string& filename) {
 // Get the absolute path of this protoc binary.
 bool GetProtocAbsolutePath(string* path) {
 #ifdef _WIN32
-  char buffer[MAX_PATH];
-  int len = GetModuleFileName(NULL, buffer, MAX_PATH);
+  wchar_t wide_buffer[MAX_PATH];
+  // Note: the returned length does not include terminating null character.
+  int len = GetModuleFileNameW(NULL, wide_buffer, MAX_PATH);
+  if (len == GOOGLE_ARRAYSIZE(wide_buffer) &&
```

Comment:
Unfortunately No.  This CL only adds error checking.  Supporting non-ASCII characters requires more non-trivial changes hence it is beyond the goal of this PR.

Basically we need to use UTF-16 version of APIs (APIs end with 'W') everywhere. Probably it also requires us to convert UTF-16 string into UTF-8 string around those API calls so that string operations can be shared with other platforms (Linux, OS X, and so on).